src/ctap/mod.rs

```diff
@@ -751,14 +754,37 @@ where
 
     fn process_get_info(&self) -> Result<ResponseData, Ctap2StatusCode> {
         let mut options_map = BTreeMap::new();
-        // TODO(kaczmarczyck) add FIDO 2.1 options
+        // TODO(kaczmarczyck) add authenticatorConfig and credProtect options
         options_map.insert(String::from("rk"), true);
         options_map.insert(String::from("up"), true);
         options_map.insert(
             String::from("clientPin"),
             self.persistent_store.pin_hash().is_some(),
         );
         Ok(ResponseData::AuthenticatorGetInfo(
+            #[cfg(feature = "with_ctap2_1")]
+            AuthenticatorGetInfoResponse {
+                versions: vec![
+                    #[cfg(feature = "with_ctap1")]
+                    String::from(U2F_VERSION_STRING),
+                    String::from(FIDO2_VERSION_STRING),
+                ],
+                extensions: Some(vec![String::from("hmac-secret")]),
+                aaguid: *AAGUID,
+                options: Some(options_map),
+                max_msg_size: Some(1024),
+                pin_protocols: Some(vec![
+                    CtapState::<R, CheckUserPresence>::PIN_PROTOCOL_VERSION,
+                ]),
+                max_credential_count_in_list: MAX_CREDENTIAL_COUNT_IN_LIST.map(|c| c as u64),
+                // You can use ENCRYPTED_CREDENTIAL_ID_SIZE here, but if your
+                // browser passes that value, it might be used to fingerprint.
+                max_credential_id_length: None,
+                transports: Some(vec![AuthenticatorTransport::Usb]),
+                algorithms: Some(vec![ES256_CRED_PARAM]),
+                firmware_version: None,
```

Comment:
As an alternative to duplicating all the other fields, these ones could be individually enabled by `cfg`. This would also be consistent with the definition of the struct in the other file.

Reply:
Done.